docs/architecture/ADR-002-deposit-assets.md

```diff
@@ -0,0 +1,118 @@
+# ADR 002: Depositing assets for uTokens
+
+## Changelog
+
+- September 13, 2021: Initial Draft (@toteki)
+
+## Status
+
+Proposed
+
+## Context
+
+One of the base functions of the Umee universal capital facility is to allow liquidity providers to deposit assets, and earn interest on their deposits.
+
+The associated feature “Lender deposits asset for uToken & redeems uToken for a single cosmos asset type" was initially discussed as follows:
+- Lender deposits (locks) a cosmos asset type (likely Atoms or uumee) into asset facilities
+- Facility mints and sends u-Assets in response (u-Atom, u-uumee)
+- Lender redeems u-Assets for original assets
+- Asset facility knows its current balances of all asset types
+- Asset facility knows the amount of all uToken types in circulation
+
+The Cosmos `x/bank` module can be used as the basis for these capabilities.
+
+Note that the exchange rate of Assets:uAssets will be a shifting exchange rate that grows with interest - see ADR-001.
+
+## Alternative Approaches
+
+While the proposed implementation will use the Cosmos banking module to simultaneously transfer assets and mint uTokens, it might also be possible to use a [Liquidity Pool](https://tutorials.cosmos.network/liquidity-module/) for the non-minting portion. The capital facility would be just another account which offers trades on Asset:uAsset pools (though it would still require a separate way of minting the uTokens it offers). This option should be considered if direct use of the banking module is rejected.
```

Comment:
```suggestion
While the proposed implementation will use the Cosmos banking module to simultaneously transfer assets and mint uTokens, it might also be possible to use a [Liquidity Pool](https://tutorials.cosmos.network/liquidity-module/) for the non-minting portion.
The capital facility would be just another account which offers trades on Asset:uAsset pools (though it would still require a separate way of minting the uTokens it offers). This option should be considered if direct use of the banking module is rejected.
```